compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1246,7 +1252,8 @@ impl Context {
             let read = self.acir_context.read_from_memory(source, &index_var)?;
             Ok::<AcirValue, RuntimeError>(AcirValue::Var(read, AcirType::field()))
         })?;
-        self.initialize_array(destination, array_len, Some(AcirValue::Array(init_values.into())))?;
+        let array: im::Vector<AcirValue> = init_values.into();
+        self.initialize_array(destination, array_len, Some(AcirValue::Array(array.clone())))?;
```

Comment:
```suggestion
        self.initialize_array(destination, array_len, Some(AcirValue::Array(array)))?;
```
nit: I don't think we need this